scripts/bundle-rhmi-operators.sh

```diff
@@ -56,24 +57,23 @@ copy_bundles() {
   done
 }
 
-# Remove the replaces field in the csv to allow for a single bundle install. i.e.
+# Remove the replaces field in the csv to allow for a single bundle install or an upgrade install. i.e.
 # The install will not require a previous version to replace.
 check_upgrade_install() {
   if [ "$UPGRADE_RHMI" = true ] ; then
     # We can return as the csv will have the replaces field by default
-    echo 'Not replacing rhmi operator'
+    echo 'Not removing replaces field in CSV'
     return
   fi
-
-  echo "versions:::: "$VERSIONS
   # Get the oldest version, example: VERSIONS="2.5,2.4,2.3" oldest="2.3"
   OLDEST_VERSION=${VERSIONS##*,}
 
-  printf "Removing replaces field from CSV \n"
-  file=`ls './'$OLDEST_VERSION | grep .clusterserviceversion.yaml`
-  sed '/replaces/d' './'$OLDEST_VERSION'/'$file > newfile ; mv newfile './'$OLDEST_VERSION'/'$file
+  file=("./$(echo ${OLDEST_VERSION} | sed "s/,/ /g")/$(echo ${OLM_TYPE} | sed "s/,/ /g").v$(echo ${OLDEST_VERSION} | sed "s/,/ /g").clusterserviceversion.yaml")
```

Comment:
This won't work with manifests generated with the new version of the operator-sdk, since the filename of the CSV now is  `managed-api-service.clusterserviceversion.yaml` (missing the release version that is assumed in this line).

To test this scenario, generate new manifests and run the script for that release:
```sh
SEMVER=1.3.0 OLM_TYPE=managed-api-service make release/prepare
OC_INSTALL=false ORG=<org> OLM_TYPE=managed-api-service BUNDLE_VERSIONS=1.3.0 make create/olm/bundle
```

It fails the validation as the replaces field is not removed. That's why I modified that line in the migration commit to use grep, so it works for both pre-migration and post-migration manifests.

Reply:
Makes sense, going to rollback the changes here! thanks @sergioifg94